service-csw/src/main/java/fi/nls/oskari/csw/helper/CSWISORecordDataQualityParser.java

```diff
@@ -203,29 +212,33 @@ public CSWISORecordDataQualityParser() {
         Node quantitativeResultValueUnitNode = (Node) XPATH_QUANTITATIVE_RESULT_VALUE_UNIT.evaluate(parentNode, XPathConstants.NODE);
         dataQualityObjectQuantitativeResult.setValueUnit(localize(quantitativeResultValueUnitNode));
 
-        NodeList quantitativeResultErrorStatisticNode = (NodeList) XPATH_QUANTITATIVE_RESULT_ERROR_STATISTIC.evaluate(parentNode, XPathConstants.NODESET);
-        List<String> valueValueList = new ArrayList<>();
-        for (int i = 0;i < quantitativeResultErrorStatisticNode.getLength(); ++i) {
-            valueValueList.add(localize(quantitativeResultErrorStatisticNode.item(i)));
+        NodeList quantitativeValueNodeList = (NodeList) XPATH_QUANTITATIVE_RESULT_VALUE.evaluate(parentNode, XPathConstants.NODESET);
+        List<String> valueList = new ArrayList<>();
+        for (int i = 0;i < quantitativeValueNodeList.getLength(); ++i) {
+            valueList.add(localize(quantitativeValueNodeList.item(i)));
         }
-        dataQualityObjectQuantitativeResult.setValue(valueValueList);
+        dataQualityObjectQuantitativeResult.setValue(valueList);
 
-        Node quantitativeValueNode = (Node) XPATH_QUANTITATIVE_RESULT_VALUE.evaluate(parentNode, XPathConstants.NODE);
-        dataQualityObjectQuantitativeResult.setErrorStatistic(localize(quantitativeValueNode));
+        Node quantitativeResultErrorStatisticNode = (Node) XPATH_QUANTITATIVE_RESULT_ERROR_STATISTIC.evaluate(parentNode, XPathConstants.NODE);
+        dataQualityObjectQuantitativeResult.setErrorStatistic(localize(quantitativeResultErrorStatisticNode));
 
         return dataQualityObjectQuantitativeResult;
     }
 
     //Move to common utility class
     private String localize(final Node elem) {
-        String ret = getText(elem);
+        String ret = null;
         String localized;
-        if (elem != null && pathToLocalizedValue != null) {
+        if (elem != null) {
             try {
-                final Node localeNode = (Node) pathToLocalizedValue.evaluate(elem, XPathConstants.NODE);
-                localized  = getText(localeNode);
-                if (localized != null && !localized.isEmpty()) {
-                    ret = localized;
+                Node contentNode = (Node) XPATH_CHARACTER_STRING.evaluate(elem, XPathConstants.NODE);
+                ret = getText(contentNode);
+                if (pathToLocalizedValue != null){
```

Comment:
The logic here is to use the text content of `../gmd:LI_Lineage/gmd:statement` node as a fallback value the node doesn't have `../gmd:PT_FreeText/gmd:textGroup/gmd:LocalisedCharacterString[@locale='#foo']` doesn't exist? Is that correct? Obviously if it does have a <gmd:PT_FreeText> element as it's child it can not have a text content, right? Then I'd argue that it would be cleaner to separate the two options more clearly.

Reply:
The logic is to use non-localized content (CharacterString) if localized content is not found or it's empty. I don't know if it's wanted, but also CSWISORecordParser works similarly. This feature has been noted in the future improvements (should we return empty string or non-localized if localized content isn't found).

         <gmd:statement xsi:type="gmd:PT_FreeText_PropertyType"> 
               <gco:CharacterString>Kuntajako...</gco:CharacterString> 
               <gmd:PT_FreeText> 
                    <gmd:textGroup> 
                         <gmd:LocalisedCharacterString locale="#EN" >Municipal...</gmd:LocalisedCharacterString> 
                     </gmd:textGroup> 
                </gmd:PT_FreeText> 
          </gmd:statement>